src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt

```diff
@@ -0,0 +1,529 @@
+          .jsonPath("content[2].allocationId").isEqualTo(courseAllocations[2].offenderProgramReferenceId)
+      }
+
+      @Test
+      fun `finds the second page of allocations`() {
+        val courseActivities = mutableListOf<CourseActivity>()
+        val courseAllocations = mutableListOf<OffenderProgramProfile>()
+        val pageSize = 3
+        nomisDataBuilder.build {
+          programService {
+            repeat(pageSize + 1) {
+              courseActivities += courseActivity(startDate = "$today")
+            }
+          }
+          offender {
+            booking {
+              courseActivities.forEach {
+                courseAllocations += courseAllocation(courseActivity = it, startDate = "$today")
+              }
+            }
+          }
+        }
+
+        webTestClient.getActiveAllocations(pageSize = pageSize, page = 1)
+          .expectBody()
+          .jsonPath("content.size()").isEqualTo(1)
+          .jsonPath("content[0].allocationId").isEqualTo(courseAllocations[3].offenderProgramReferenceId)
+      }
+
+      @Test
+      fun `should return correct paging details for all pages`() {
+        val courseActivities = mutableListOf<CourseActivity>()
+        nomisDataBuilder.build {
+          programService {
+            repeat(50) {
+              courseActivities += courseActivity(startDate = "$today")
+            }
+          }
+          courseActivities.forEachIndexed { index, activity ->
+            offender {
+              booking {
+                // give half of the activities active allocations
+                if (index % 2 == 0) {
+                  courseAllocation(courseActivity = activity, startDate = "$today")
+                }
+                // and half of the activities inactive allocations
+                else {
+                  courseAllocation(courseActivity = activity, startDate = "$yesterday", endDate = "$yesterday")
+                }
+              }
+            }
+          }
+        }
+
+        webTestClient.getActiveAllocations(page = 0)
+          .expectBody()
+          .jsonPath("totalElements").isEqualTo(25)
+          .jsonPath("numberOfElements").isEqualTo(10)
+          .jsonPath("number").isEqualTo(0)
+          .jsonPath("totalPages").isEqualTo(3)
+          .jsonPath("size").isEqualTo(10)
+          .jsonPath("content.length()").isEqualTo(10)
+
+        webTestClient.getActiveAllocations(page = 1)
+          .expectBody()
+          .jsonPath("numberOfElements").isEqualTo(10)
+          .jsonPath("number").isEqualTo(1)
+          .jsonPath("content.length()").isEqualTo(10)
+
+        webTestClient.getActiveAllocations(page = 2)
+          .expectBody()
+          .jsonPath("numberOfElements").isEqualTo(5)
+          .jsonPath("number").isEqualTo(2)
+          .jsonPath("content.length()").isEqualTo(5)
+      }
+    }
+
+    @Nested
+    inner class AllocationSelection {
```

Comment:
The new course allocation finder selection tests are almost identical to the course activity tests - again how can we stop these diverging?